nginx-controller/nginx/templates/nginx.ingress.tmpl

```diff
@@ -5,6 +5,32 @@ upstream {{$upstream.Name}} {
 	server {{$server.Address}}:{{$server.Port}};{{end}}
 }{{end}}
 
+{{with .EndpointsInfo}}
+map $upstream_addr $namespace_name {
```

Comment:
This won't work if you have more than one Ingress resource deployed. In that case you will get several maps generated, which will collide with each other.
The way to solve that problem is to prefix the variable name with the namespace and the name of the corresponding Ingress resource:
```nginx
map $upstream_addr $IngressNameSpace_IngressName_namespace_name {
...
}
```
However, this will require to use `set`:
```nginx
 set $namespace_name $IngressNameSpace_IngressName_namespace_name;
```

Also, I suggest using the following variable names:
```
$k8s_upstream_pod
$k8s_upstream_container
$k8s_upstream_namespace
$k8s_upstream_pod_labels
```
Note the prefix `k8s` for the k8s-related variables.